CRTX-180889-Synopsys-Coverity (#40641)
demisto/content: Packs/SynopsysCoverity/README.md

# Synopsys Coverity

This pack includes Cortex XSIAM parsing and modeling rules (XDM schema) for Synopsys Coverity enhanced usage logging.

> 

Reviewer: Lets also add these in the beginning (As available and needed):
## This pack includes:
 
Data normalization capabilities:
  * Rules for <parsing_and_modeling> <vendor's log type> logs that are ingested via the <ingestion_method> on Cortex XSIAM.
    * The ingested <vendor_product> logs can be queried in XQL Search using the *`<dataset_name>`* dataset.
 
## Supported log categories
| Category                    | Category Display Name                 |
|:----------------------------|:--------------------------------------|
| Category_1 <link_if_exist>  | Category 1                            |
| Category_2 <link_if_exist>  | Category 2                            |
 
### Supported Timestamp Formats:
<Enter time format when pack contains time parsing>

# Synopsys Coverity

> This pack includes XSIAM parsing and modeling rules (XDM schema) for Synopsys Coverity Enhanced Usage Logging.

Reviewer 2: ```suggestion
This pack includes Cortex XSIAM parsing and modeling rules (XDM schema) for Synopsys Coverity enhanced usage logging.
```

## Configuration on Server Side

> Coverity does not support native log forwarding (e.g., Syslog). Instead, it writes enhanced usage logs to local files on the host filesystem.

Reviewer 2: ```suggestion
Synopsis Coverity does not support native log forwarding (for example, Syslog). Instead, it writes enhanced usage logs to local files on the host filesystem.
```

>     1. Ensure that Coverity is configured to generate Enhanced Usage Logs.

Reviewer 2: ```suggestion
    1. Ensure that Synopsis Coverity is configured to generate enhanced usage logs.
```

> 

Reviewer 2: ```suggestion
```

>     2. The log files are typically written under the Coverity installation directory, usually in: `<install_dir>\logs`

Reviewer 2: ```suggestion
      The log files are typically written under the Synopsis Coverity installation directory, usually under `<install_dir>\logs`
```

> 

Reviewer 2: ```suggestion
```

>     3. Verify that the logs are being written in **JSON format** (e.g., coverity_usage-*.log).

Reviewer 2: ```suggestion
    2. Verify the logs are being written in JSON format (for example, coverity_usage-*.log).
```

> If logs are not found in the default path, refer to [Coverity Enhanced Usage Logging documentation](https://documentation.blackduck.com/bundle/coverity-docs/page/coverity-platform/topics/enhanced_usage_logging.html) for exact location and format details.

Reviewer 2: ```suggestion
If logs are not found in the default path, see the [Synopsis Coverity enhanced usage logging documentation](https://documentation.blackduck.com/bundle/coverity-docs/page/coverity-platform/topics/enhanced_usage_logging.html) for exact location and format details.
```

### Filebeat Collection

> In order to collect the logs and forward them to Cortex XSIAM, use the following collector:

Reviewer 2: ```suggestion
To collect the logs and forward them to Cortex XSIAM, use the XDRC (XDR Collector).
```

> 

Reviewer 2: ```suggestion
```

> #### XDRC (XDR Collector)

Reviewer 2: ```suggestion
```

> 

Reviewer 2: ```suggestion
```

> You will configure a profile for the XDR Collector and assign the correct vendor and product values.

Reviewer 2: ```suggestion
1. Configure a profile for the XDR Collector and assign the correct vendor and product values.
```

> 

Reviewer 2: ```suggestion
```

To set up the collector, follow the instructions outlined in the official [Cortex XSIAM XDR Collector documentation](https://docs.paloaltonetworks.com/cortex/cortex-xdr/cortex-xdr-pro-admin/cortex-xdr-collectors/xdr-collector-datasets#id7f0fcd4d-b019-4959-a43a-40b03db8a8b).

> The dataset name will be:

Reviewer 2: ```suggestion
The dataset name will be `synopsys_coverity_raw`
```

> 

Reviewer 2: ```suggestion
```

> `synopsys_coverity_raw`

Reviewer 2: ```suggestion
```
